src/charm.py

```diff
@@ -370,7 +370,10 @@ def _on_postgresql_pebble_ready(self, event: WorkloadEvent) -> None:
         # Otherwise, each unit will create a different cluster and
         # any update in the members list on the units won't have effect
         # on fixing that.
-        if not self.unit.is_leader() and "cluster_initialised" not in self._peers.data[self.app]:
+        # Also defers when the workload is not reachable.
+        if (
+            not self.unit.is_leader() and "cluster_initialised" not in self._peers.data[self.app]
+        ) or not container.can_connect():
```

Comment:
I believe the container must be available for `pebble_ready` to be emitted, so I'm not sure this check is necessary

Reply:
The harness threw an connection error while testing. Not sure if it is misbehaviour or a possible IRL issue, so added the check just in case.

Reply:
In which specific test the error happens? I believe it should work the way Will commented. Otherwise we can file a bug on the operator framework repo.

Reply:
You are correct. I rechecked and the ConnectionError was thrown by harness.get_container_pebble_plan, not by the charm so there is no need to check there.